modules/edu.gemini.seqexec.model/src/main/scala/edu/gemini/seqexec/model/SharedModel.scala

```diff
@@ -47,7 +48,7 @@ object SharedModel {
     case class Running(progress: Double) extends ActionStatus
   }
 
-  trait Step {
+  sealed trait Step {
```

Comment:
Are open traits ever used as when doing idiomatic FP in Scala?

It seems to me that the idiomatic way to use sealed traits is when there are a finite number of case classes like here. Otherwise a *type class* pattern is used (like in `simulacrum`). Is my assumption correct?